app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/ComposeActivity.kt

```diff
@@ -54,19 +72,62 @@ class ComposeActivity : ComponentActivity() {
         super.onCreate(savedInstanceState)
         setContent {
             val navController = rememberNavController()
+            val webViewNavigator = rememberWebViewNavigator()
+            val scaffoldState = rememberScaffoldState()
+            val scope = rememberCoroutineScope()
 
             ArchitectureDemoTheme {
                 Scaffold(
+                    scaffoldState = scaffoldState,
                     topBar = {
-                        if (activityViewModel.showToolbar.collectAsState().value) {
-                            TopAppBar {
-                                Text(activityViewModel.displayName.collectAsState().value)
-                            }
+                        AnimatedVisibility(
+                            visible = activityViewModel.showToolbar.collectAsState(false).value,
```

Comment:
Easy entrance exit animations.  We can even float up the definition of certain app bar elements and then animate then them individually with Modifier.enterExitAnim().